pkg/discovery/worker/group_discovery_worker.go

```diff
@@ -257,9 +257,6 @@ func (worker *k8sEntityGroupDiscoveryWorker) createSLOPolicy(
 	}
 	var settingValue float32
 	for _, setting := range spec.Objectives {
-		if err := json.Unmarshal(setting.Value.Raw, &settingValue); err != nil {
-			return nil, err
-		}
 		switch setting.Name {
 		case v1alpha1.ResponseTime:
 			settings.AddSetting(group.NewResponseTimeSLOPolicySetting(settingValue))
```

Comment:
Isn't `settingValue` not yet initialized here?

Reply:
I also dont understand this change, why add a zero setting value?

Reply:
Oh, I see where the confusion is, the following logic is moved to line 268:

```
		if err := json.Unmarshal(setting.Value.Raw, &settingValue); err != nil {
			return nil, err
		}
```